server/src/internalClusterTest/java/org/elasticsearch/persistent/PersistentTaskCreationFailureIT.java

```diff
@@ -0,0 +1,227 @@
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(FailingCreationPersistentTasksPlugin.class);
+    }
+
+    private static boolean hasPersistentTask(ClusterState clusterState) {
+        return findTasks(clusterState, FailingCreationPersistentTaskExecutor.TASK_NAME).isEmpty() == false;
+    }
+
+    public void testPersistentTasksThatFailDuringCreationAreRemovedFromClusterState() {
+
+        final var masterClusterService = internalCluster().getCurrentMasterNodeInstance(ClusterService.class);
+        final var plugins = StreamSupport.stream(internalCluster().getInstances(PluginsService.class).spliterator(), false)
+            .flatMap(ps -> ps.filterPlugins(FailingCreationPersistentTasksPlugin.class))
+            .toList();
+        plugins.forEach(plugin -> plugin.hasFailedToCreateTask.set(false));
+
+        final var taskCreatedListener = ClusterServiceUtils.addTemporaryStateListener(
+            masterClusterService,
+            PersistentTaskCreationFailureIT::hasPersistentTask
+        );
+
+        taskCreatedListener.andThenAccept(v -> {
+            // enqueue some higher-priority cluster state updates to check that they do not cause retries of the failing task creation step
+            for (int i = 0; i < 5; i++) {
+                masterClusterService.submitUnbatchedStateUpdateTask("test", new ClusterStateUpdateTask(Priority.IMMEDIATE) {
+                    @Override
+                    public ClusterState execute(ClusterState currentState) {
+                        assertTrue(hasPersistentTask(currentState));
+
+                        assertTrue(waitUntil(() -> {
+                            final var completePersistentTaskPendingTasksCount = masterClusterService.getMasterService()
+                                .pendingTasks()
+                                .stream()
+                                .filter(
+                                    pendingClusterTask -> pendingClusterTask.getSource().string().equals("finish persistent task (failed)")
+                                )
+                                .count();
+                            assertThat(completePersistentTaskPendingTasksCount, lessThanOrEqualTo(1L));
+                            return completePersistentTaskPendingTasksCount == 1L;
+                        }));
+
+                        return currentState.copyAndUpdateMetadata(
+                            mdb -> mdb.putCustom(
+                                PersistentTasksCustomMetadata.TYPE,
+                                PersistentTasksCustomMetadata.builder(
+                                    PersistentTasksCustomMetadata.getPersistentTasksCustomMetadata(currentState)
+                                )
+                                    // create and remove a fake task just to force a change in lastAllocationId
+                                    .addTask("test", "test", null, PersistentTasksCustomMetadata.INITIAL_ASSIGNMENT)
```

Comment:
why do we need to force a change?

Reply:
To avoid the fast-path through `PersistentTasksNodeService` that avoids the retry bug if this bit of the metadata is unchanged:

https://github.com/elastic/elasticsearch/blob/2380778bc175857c385422d2c6026e7f41a357c4/server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java#L108